File: C++/level1/정수_내림차순으로_배치하기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

long long solution(long long n) {
    long long answer = 0;
    vector<int> numbers;
    
    while(n != 0){
        numbers.push_back(n%10);
        n = n/10;
    }
  
    sort(numbers.begin(), numbers.end());

    while(!numbers.empty()){
        answer = answer*10 + numbers.back();
        numbers.pop_back();
    }

    return answer;
}

File: JavaScript/level1/정수_내림차순으로_배치하기.js
function solution(n) {
    var answer = 0;
    let array = [];
    
    while(n != 0){
        array.push(n%10);
        n = (n - n%10)/10;
    }

    array.sort().reverse();
    
    for(let i of array)
        answer =  answer*10 + i;
    
    return answer;
}
